examples/electron_tests.html

```diff
@@ -0,0 +1,442 @@
+<!doctype html>
+<html>
+<body>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/three.js/95/three.js"></script>
+<script src="https://threejs.org/examples/js/controls/OrbitControls.js"></script>
+<script>
+
+  const width = 1024;
+  const height = 513;
+  let hasWarned,hasStartedTests;
+  console.warn = function(){
+    hasWarned = true;
+  };
+
+  let currentMin,currentMag,currentRes;
+
+  const resolutions = [
+    {x:1024,y:513},
+    {x:1024,y:512},
+  ];
+
+  const textureMagFilters = [
+    THREE.LinearFilter,
+    THREE.NearestFilter
+  ];
+
+  const textureMinFilters = [
+    THREE.NearestFilter,
+    THREE.NearestMipMapNearestFilter,
+    THREE.NearestMipMapLinearFilter,
+    THREE.LinearFilter,
+    THREE.LinearMipMapNearestFilter,
+    THREE.LinearMipMapLinearFilter
+  ];
+
+  let camera, scene, renderer, controls, browserPlane,
+    canvas, ctx, texture, browserWindow,
+    mouse = new THREE.Vector2(),
+    raycaster = new THREE.Raycaster();
+
+  init();
+  animate();
+
+  function init() {
+    setupScene();
+    setupCanvasAndTexture();
+    setupElectron()
+      .then(()=>{
+        setupBrowserPlane();
+        setupListeners();
+        return setupFilterTests();
+      });
+  }
+  async function setupFilterTests(){
+    // loop through all filter options to see it update.
+
+    let totalTests = resolutions.length * textureMinFilters.length * textureMagFilters.length;
+
+    for(let i = 0;i<resolutions.length; i++) {
+
+      for (let j = 0; j < textureMagFilters.length; j++) {
+
+        for (let k = 0; k < textureMinFilters.length; k++) {
+
+          let testCount = (i+1) * (j+1) * (k+3);
+
+          await new Promise(resolve =>
+            document.addEventListener( 'keyup', () => {
```

Comment:
I think we need to remove this listener once we hit the event, or else subsequent loop iterations will trigger this, which becomes confusing.